lib/fluent/engine.rb

```diff
@@ -209,11 +209,11 @@ def run
         $log.error_backtrace
       ensure
         $log.info "shutting down fluentd"
-        shutdown
         if @log_emit_thread
           @log_event_loop_stop = true
           @log_emit_thread.join
         end
+        shutdown
```

Comment:
Does calling `shutdown` after log thread stop cause log inside `shutdown` is dropped?